provider/internal/util/sliceutil/slice.go

```diff
@@ -23,3 +27,36 @@ func MapWithError[T any, V any](sliceT []T, mapFunc func(T) (V, error)) ([]V, er
 	}
 	return sliceV, nil
 }
+
+func StringsEqual(s1 []string, s2 []string) bool {
+	if s1 == nil && s2 == nil {
+		return true
+	}
+	if s1 == nil || s2 == nil {
```

Comment:
I think the first two condition checks are redundant as they are already covered by the next conditions and can be eliminated

Reply:
I was thinking to differentiate a nil pointer vs an empty slice, but yea, they are kind of semantically equal in this context, changed